tests/projects/test_projects.py

```diff
@@ -26,6 +26,12 @@ def _assert_dirs_equal(expected, actual):
     assert len(dir_comparison.funny_files) == 0
 
 
+def _build_uri(base_uri, subdirectory):
+    if subdirectory != "":
+        return "%s#%s" % (base_uri, subdirectory)
+    return base_uri
+
+
 def test_fetch_project(tmpdir):
```

Comment:
Refactored the _fetch_project a bit for brevity / since the `_fetch_project` API changed in this PR